File: Coding Exercises/Java/Day01/SmallestElement.java
public class Exercise {
  public static int findSmallestElement(int[] arr) {
    int smallest = Integer.MAX_VALUE;

    for (int i = 0; i < arr.length; i++) {
      if (arr[i] < smallest) {
        smallest = arr[i];
      }
    }

    return smallest;
  }
}

File: Coding Exercises/TypeScript/Day01/SmallestElement.ts
function findSmallestElement(arr: number[]): number {
    let smallest: number = Number.POSITIVE_INFINITY;

    for (let num of arr) {
        if (num < smallest) {
            smallest = num;
        }
    }

    return smallest;
}

function findSmallestElement(arr: number[]): number {
    return arr.reduce((smallest, num) => num < smallest ? num : smallest, Number.POSITIVE_INFINITY);
}